---
title: "Gene Expression Plasticity"
author: "Matthew Berginski"
date: "3/23/2020"
output: github_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(here)
library(broom)
library(tictoc)
library(glue)
library(kableExtra)

library(BerginskiRMisc)
library(DarkKinaseTools)

knitr::opts_chunk$set(include = TRUE, echo = F, warning = F, fig.align = 'center')
```

```{r read in data, cache=TRUE, include=FALSE}
process_DESeq_for_correlations <- function(file, cell_line) {
	DESeq_fold_change = read_csv(file, col_types = cols(
		hgnc_symbol = col_character(),
		treatment = col_character(),
		baseMean = col_double(),
		log2FoldChange = col_double(),
		lfcSE = col_double(),
		pvalue = col_double(),
		padj = col_double()
	)) %>%
		filter(!str_detect(treatment,"^batch")) %>%
		mutate(cell_line_treatment = paste0(cell_line,"_",treatment)) %>%
		select(hgnc_symbol,cell_line_treatment,log2FoldChange)
	
	return(DESeq_fold_change)
}

fold_change = bind_rows(
	process_DESeq_for_correlations(here('DESeq_results/SUM159_only_DESeq2_results.csv.gz'),"SUM159"),
	process_DESeq_for_correlations(here('DESeq_results/SUM229_only_DESeq2_results.csv.gz'),"SUM229"),
	process_DESeq_for_correlations(here('DESeq_results/HCC1806_only_DESeq2_results.csv.gz'),"HCC1806"),
	process_DESeq_for_correlations(here('DESeq_results/MDAMB231_only_DESeq2_results.csv.gz'),"MDAMB231")
)

#The fold change data doesn't necessarily contain all of the genes since we
#remove the especially low expression genes from DESeq analysis. This set of
#removed genes are by defintion low fold change genes in their corresponding
#treatment sets, so we'll set all of them to 0.
all_gene_combos = crossing(
	hgnc_symbol = unique(fold_change$hgnc_symbol), 
	cell_line_treatment = unique(fold_change$cell_line_treatment))

missing_genes = setdiff(all_gene_combos, 
												fold_change %>% select(-log2FoldChange)) %>%
	mutate(log2FoldChange = 0)

fold_change = bind_rows(fold_change, missing_genes)

fold_change = fold_change %>%
	extract(cell_line_treatment,c("cell_line","compound"), 
					"(.*)_treatment_(.+)_vs_DMSO",
					remove = F)

ontology_sets = msigdbr::msigdbr() %>%
	filter(gs_cat == "C5", gs_subcat == "GO:BP" | gs_subcat == "GO:MF")

GO_ion_channels = ontology_sets %>% filter(gs_name == "GO_CATION_CHANNEL_ACTIVITY")
GO_kinases = ontology_sets %>% filter(gs_name == "GO_PROTEIN_KINASE_ACTIVITY")
GO_GPCR = ontology_sets %>% filter(gs_name == "GO_G_PROTEIN_COUPLED_RECEPTOR_ACTIVITY")
```

```{r, include = FALSE, cache = T}
fold_change_variation = fold_change %>%
	group_by(hgnc_symbol) %>%
	summarise(gene_fold_mean = mean(log2FoldChange, na.rm = T),
						gene_fold_abs_mean = mean(abs(log2FoldChange), na.rm = T),
						gene_fold_sd = sd(log2FoldChange, na.rm = T)) %>%
	mutate(gene_fold_mean_rank = percent_rank(abs(gene_fold_mean)),
				 gene_fold_abs_mean_rank = percent_rank(gene_fold_abs_mean),
				 gene_fold_sd_rank = percent_rank(gene_fold_sd)) %>%
	mutate(protein_class = case_when(
		hgnc_symbol %in% all_kinases$symbol ~ "kinase",
		hgnc_symbol %in% GO_ion_channels$human_gene_symbol ~ "ion",
		hgnc_symbol %in% GO_GPCR$human_gene_symbol ~ "GPCR",
		TRUE ~ "other")) %>%
	mutate(is_kinase = case_when(
		hgnc_symbol %in% all_kinases$symbol ~ TRUE,
		TRUE ~ FALSE)) %>%
	arrange(desc(gene_fold_abs_mean_rank)) %>%
	identity()

protein_class = fold_change_variation %>%
	group_by(protein_class) %>%
	summarise(mean_abs_fold = mean(abs(gene_fold_mean))) %>%
	arrange(mean_abs_fold)

fold_change_variation$protein_class = fct_relevel(fold_change_variation$protein_class,
																									protein_class$protein_class)

fold_change_var_kinases = fold_change_variation %>%
	filter(hgnc_symbol %in% all_kinases$symbol) %>% 
	left_join(all_kinases %>% select(symbol,class), by=c('hgnc_symbol'='symbol'))

fold_change_var_DK = fold_change_variation %>%
	filter(hgnc_symbol %in% dark_kinases$symbol)

protein_class = fold_change_variation %>%
	group_by(protein_class) %>%
	summarise(mean_abs_fold = mean(abs(gene_fold_mean)))
```

What is gene plasticity? How about we define it as changes in expression due to perturbation. Great, we've got a pile of gene expression data in the context of drug perturbations. So far, we've got `r length(unique(fold_change$cell_line_treatment))` combinations of drug (`r length(unique(fold_change$compound))`) and cell lines (`r length(unique(fold_change$cell_line))`) covered and included.

Specifically, we have the results of the DESeq2-based gene expression analysis. The primary result of here are the log 2 fold change results for each gene. This summarizes how much the gene expression changed in comparison with the corresponding DMSO control. From this set of results, I can see a few ways to measure the plasticity of these gene responses:

* Average of Fold Changes: Simple. What's the average value of the fold changes for this gene, high values indicate a high plasticity gene.

* Average of Absolute Value of the Fold Changes: Bit more complicated. Deals with the case when a gene's expression is variable, but the direction of variation changes from condition to condition. In this case, a high fold change value in one treatment could be cancelled out by a low fold change in another treatment. I'd still say this gene has high plasticity though.

For both of these metrics, I've also collected the rank of every gene, thinking that the raw fold change summary values aren't in and of themselves that informative.

Finally, I'll also grabbed a list of Ion Channels and GPCRs from the GO ontology to make some comparisons to our sibling IDG groups.


# Distribution of Mean Log Fold Changes

Here's what the distribution of average log fold changes look like for every gene split by protein class. This plot should be read as a histogram normalized to deal with the differing number of genes in the classes. 

Every gene class is centered at zero, as expected, but the biggest take away here is that the kinases have the fewest genes in the minimal change category (lowest peak in near zero changes section of the plot)

```{r}
ggplot(fold_change_variation, aes(x=gene_fold_mean, y = after_stat(density), color=protein_class)) + 
	geom_freqpoly(bins=60, lwd=1, alpha=0.75) +
	labs(x="Mean Log 2 Fold Change in Expression",y="Density of Genes") +
	BerginskiRMisc::theme_berginski()
```

Same data as above, except unstacking the histograms.

```{r}
library(ggridges)
ggplot(fold_change_variation, aes(x = gene_fold_mean, y = protein_class)) + 
	geom_density_ridges(bandwidth = 0.03) +
	labs(x="Mean Log 2 Fold Change in Expression",y="Protein Class") +
	BerginskiRMisc::theme_berginski()
```

# Distribution of Mean ABS Fold Change

Here's the same style normalized histogram for the distribution of the mean of the absolute value of fold change in expression.

GPCRs really don't change their expression levels in our treatments, do they? Otherwise, kinases have the highest average values on this metric as well.

```{r}
ggplot(fold_change_variation, aes(x=gene_fold_abs_mean, y = after_stat(density), color=protein_class)) + 
	geom_freqpoly(bins=30,lwd=1) +
	xlim(c(0,NA)) +
	labs(x="Mean ABS Log 2 Fold Change in Expression",y="Density of Genes") +
	BerginskiRMisc::theme_berginski()
```

Same data as above with the histograms spread out.

```{r}
library(ggridges)
ggplot(fold_change_variation, aes(x = gene_fold_abs_mean, y = protein_class)) + 
	geom_density_ridges(bandwidth = 0.05) +
	xlim(c(0,NA)) +
	labs(x="Mean ABS Log 2 Fold Change in Expression",y="Protein Class") +
	BerginskiRMisc::theme_berginski()
```

# Data Organized by Gene Ranks

The above numbers are only so meaningful to me though. Is a mean absolute value of 2-fold variation of 0.35 high or low? So I've also calculated the percentile rank of every gene based on the absolute value fold change average from above. High values here indicate that a gene has a higher average fold change (so 0.9 or 90% percentile gene has a higher fold change than 90% of the genome).

```{r}
ggplot(fold_change_variation, aes(y=gene_fold_abs_mean_rank, x=protein_class)) + 
	geom_boxplot() + 
	labs(y="Mean ABS Log 2 Fold Percentile",x="Protein Class") +
	theme_berginski()
```

The kinases have a stronger response here (average percentile: `r signif(mean(fold_change_var_kinases$gene_fold_abs_mean_rank),2)`) than the rest of the genome (p-value: `r tidy(t.test(gene_fold_abs_mean_rank ~ is_kinase, data=fold_change_variation))$p.value`).

# Differences Between Light and Dark Kinases

Looking only at the kinases, we can take a look at differences in ranking for the light vs dark kinases:

```{r}
ggplot(fold_change_var_kinases, aes(y=gene_fold_abs_mean_rank, x=class)) + 
	geom_boxplot() + 
	labs(y="Mean ABS Log 2 Fold Percentile",x="Kinase Class") +
	theme_berginski()
```

The light kinases are a bit higher on average than the dark kinases (`r signif(tidy(t.test(fold_change_var_kinases$gene_fold_abs_mean_rank ~ fold_change_var_kinases$class))$estimate1,3)` vs `r signif(tidy(t.test(fold_change_var_kinases$gene_fold_abs_mean_rank ~ fold_change_var_kinases$class))$estimate2,3)`). The difference in class is significant, but is only a few percent different. 

## Most Plastic Light Kinases

Here's the list of top 10 most plastic light kinases along with the genome-wide percentile ranking:

```{r}
fold_change_var_kinases %>% 
	filter(class == "Light") %>% 
	select(hgnc_symbol, gene_fold_abs_mean_rank) %>% 
	slice(1:10) %>%
	kable()
```

## Most Plastic Dark Kinases

Here's the list of top 10 most plastic light kinases along with the genome-wide percentile ranking:

```{r}
fold_change_var_kinases %>% 
	filter(class == "Dark") %>% 
	select(hgnc_symbol, gene_fold_abs_mean_rank) %>% 
	slice(1:10) %>%
	kable()
```